src/graphql/generateFlowTypes.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import prettier, {type Options as PrettierOptions} from "prettier";
+
+import type {
+  ConnectionFieldType,
+  FieldType,
+  IdFieldType,
+  NestedFieldType,
+  NodeFieldType,
+  PrimitiveFieldType,
+  Schema,
+} from "./schema";
+
+export default function generateFlowTypes(
+  schema: Schema,
+  prettierOptions: PrettierOptions
+): string {
+  const typedefs = [];
+
+  function formatField(field: FieldType): string {
+    switch (field.type) {
+      case "ID":
+        return formatIdField(field);
+      case "PRIMITIVE":
+        return formatPrimitiveField(field);
+      case "NODE":
+        return formatNodeField(field);
+      case "CONNECTION":
+        return formatConnectionField(field);
+      case "NESTED":
+        return formatNestedField(field);
+      // istanbul ignore next: unreachable per Flow
+      default:
+        throw new Error((field.type: empty));
+    }
+  }
+  function formatIdField(_unused_field: IdFieldType): string {
+    return "string";
+  }
+  function formatPrimitiveField(field: PrimitiveFieldType): string {
+    if (field.annotation == null) {
+      return "mixed";
+    } else if (field.annotation.nonNull) {
+      return field.annotation.elementType;
+    } else {
+      return "null | " + field.annotation.elementType;
+    }
+  }
+  function formatNodeField(field: NodeFieldType): string {
+    return "null | " + field.elementType;
+  }
+  function formatConnectionField(field: ConnectionFieldType): string {
+    return `$ReadOnlyArray<null | ${field.elementType}>`;
```

Comment:
Just checking my understanding: it's <null | elementType> because if the object was deleted or someting, we may still have a connection pointing to it, but won't actually be able to fetch it? (so it will be in the array as null, rather than implicitly filtered out)

Reply:
The real reason that it’s `null | elementType` is that all the
connections provided by GitHub return lists of nullable items:

  - <https://developer.github.com/v4/object/issuecommentconnection/>
  - <https://developer.github.com/v4/object/userconnection/>
  - <https://developer.github.com/v4/object/reactionconnection/>
  - etc.

Therefore, to faithfully match the schema, we have to mark these results
as nullable.

As usual, the GitHub docs do not specify _when_ such a connection might
contain a `null` item. Your explanation may be the case, but I can’t
say. I don’t believe that I’ve ever actually seen a `null` element in an
actual connection result.